src/main/scala/config/KafkaConfig.scala

```diff
@@ -76,9 +77,12 @@ object KafkaConfig {
           ("value.serializer", "org.apache.kafka.common.serialization.StringSerializer"),
           ("bootstrap.servers", broker),
           ("linger.ms", lingerTime),
-          ("max.block.ms", maxBlockMS)
+          ("max.block.ms", maxBlockMS),
+          //todo: add condition here
+          ("partitioner.class", if (useGroupPartitioner) "partitioners.GroupPartitioner" else "org.apache.kafka.clients.producer.internals.DefaultPartitioner")
```

Comment:
@omerlh I would like to not add `partitioner.class` if useGroupPartitioner=false. also didn't figure out how

Reply:
I figured this one out